datafusion/src/physical_optimizer/repartition.rs

```diff
@@ -178,38 +335,46 @@ mod tests {
             .collect()
     }
 
+    /// Runs the repartition optimizer and asserts the plan against the expected
+    macro_rules! assert_optimized {
+        ($EXPECTED_LINES: expr, $PLAN: expr) => {
+            let expected_lines: Vec<&str> = $EXPECTED_LINES.iter().map(|s| *s).collect();
+
+            // run optimizer
+            let optimizer = Repartition {};
+            let optimized = optimizer
+                .optimize($PLAN, &ExecutionConfig::new().with_target_partitions(10))?;
+
+            // Now format correctly
+            let plan = displayable(optimized.as_ref()).indent().to_string();
+            let actual_lines = trim_plan_display(&plan);
+
+            assert_eq!(
+                &expected_lines, &actual_lines,
+                "\n\nexpected:\n\n{:#?}\nactual:\n\n{:#?}\n\n",
+                expected_lines, actual_lines
+            );
+        };
+    }
+
     #[test]
     fn added_repartition_to_single_partition() -> Result<()> {
-        let optimizer = Repartition {};
+        let plan = hash_aggregate(parquet_exec());
```

Comment:
I cleaned up the tests here to reduce the ceremony of invoking the optimizer. The plans are all the same